pymodm/fields.py

```diff
@@ -1137,6 +1143,87 @@ def to_mongo(self, value):
         return [self._model_to_document(doc) for doc in value]
 
 
+class EmbeddedModelField(RelatedEmbeddedModelFieldsBase):
```

Comment:
It would be nice it both `EmbeddedDocumentField` and `EmbeddedModelField` can share the same implementation. Can you refactor these classes into a private base class such as `_EmbeddedDocumentFieldBase`?  The same can be done for `EmbeddedDocumentListField` with a `_EmbeddedDocumentListFieldBase` class.

What do you think?

Reply:
I think it's fine to put all 4 classes in this file.